src/Model/User.php

```diff
@@ -1620,16 +1620,24 @@ public static function sendRegisterOpenEmail(L10n $l10n, array $user, string $si
 	 * @param int $uid user to remove
 	 * @return bool
 	 * @throws HTTPException\InternalServerErrorException
+	 * @throws HTTPException\NotFoundException
 	 */
 	public static function remove(int $uid): bool
 	{
 		if (empty($uid)) {
-			return false;
+			throw new \InvalidArgumentException('uid needs to be greater than 0');
 		}
 
 		Logger::notice('Removing user', ['user' => $uid]);
 
-		$user = DBA::selectFirst('user', [], ['uid' => $uid]);
+		$user = self::getById($uid);
+		if (!$user) {
+			throw new HTTPException\NotFoundException('User not found with uid: ' . $uid);
+		}
+
+		if (DBA::exists('user', ['parent-uid' => $uid])) {
```

Comment:
There is an additional way to delegate, using the `manage` table.

Reply:
`User::getIdentities` returns results even for delegated accounts that can be freely deleted, making it unsuitable for this check.

I’ll add an exception for manager accounts, although I’m not sure what’s the way out, can manager accounts sever the management link?

Reply:
From my point of view we should respect all of this.

Reply:
That's not my question. When the User::remove call fails because the account is recorded as a manager for another account, can they sever the management link themselves? Should this check only done during self-removal? What about when an account is removed from the moderation section?

Reply:
I must confess that for a long term I hadn't got a look at this kind of delegation. We should test this out.

Reply:
Are you talking about the possibility to manage other accounts via the 'delegate' way? As a delegate I am not able to access the settings of a delagated account (403 error) and thus the 'remove account' dialogue isn't accessible as well.

Reply:
I know. But I had the idea that you mostly don't want to remove an account that is used to manage other accounts. And if you want to, you have to remove the access.